python/ray/train/lightning/_lightning_utils.py

```diff
@@ -0,0 +1,94 @@
+import logging
+import torch
+from typing import Any, Dict, Optional
+
+import pytorch_lightning as pl
+from pytorch_lightning.strategies import DDPStrategy
+from pytorch_lightning.plugins.environments import LightningEnvironment
+
+import ray
+from ray.air import session
+
+from torch.utils.data import IterableDataset, DataLoader
+from ray.data.dataset import DatasetIterator
+
+logger = logging.getLogger(__name__)
+
+
+class RayDDPStrategy(DDPStrategy):
+    """Subclass of DDPStrategy to ensure compatibility with Ray orchestration."""
+
+    @property
+    def root_device(self) -> torch.device:
+        return ray.train.torch.get_device()
+
+    @property
+    def distributed_sampler_kwargs(self) -> Dict[str, Any]:
+        return dict(
+            num_replicas=self.world_size,
+            rank=self.global_rank,
+        )
+
+
+class RayEnvironment(LightningEnvironment):
+    """Setup Lightning DDP training environment for Ray cluster."""
+
+    def world_size(self) -> int:
+        return session.get_world_size()
+
+    def global_rank(self) -> int:
+        return session.get_world_rank()
+
+    def local_rank(self) -> int:
+        return session.get_local_rank()
+
+    def node_rank(self) -> int:
+        return session.get_node_rank()
+
+    def set_world_size(self, size: int) -> None:
```

Comment:
it's weird to take a parameter, but then never use it.
can we omit the size parameter? or maybe:
```
def set_world_size(self, _: int) -> None:
```